cantor-archive/src/main/java/com/salesforce/cantor/archive/file/AbstractBaseArchiverOnFile.java

```diff
@@ -0,0 +1,88 @@
+package com.salesforce.cantor.archive.file;
+
+import org.apache.commons.compress.archivers.ArchiveInputStream;
+import org.apache.commons.compress.archivers.ArchiveOutputStream;
+import org.apache.commons.compress.archivers.tar.TarArchiveEntry;
+import org.apache.commons.compress.archivers.tar.TarArchiveInputStream;
+import org.apache.commons.compress.archivers.tar.TarArchiveOutputStream;
+import org.apache.commons.compress.compressors.gzip.GzipCompressorInputStream;
+import org.apache.commons.compress.compressors.gzip.GzipCompressorOutputStream;
+
+import java.io.BufferedInputStream;
+import java.io.BufferedOutputStream;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import static com.salesforce.cantor.common.CommonPreconditions.checkArgument;
+import static com.salesforce.cantor.common.CommonPreconditions.checkString;
+
+public abstract class AbstractBaseArchiverOnFile {
+    protected final String baseDirectory;
+    protected final long chunkMillis;
+
+    // default to nothing for no sub directory
+    protected String subDirectory = "";
+
+    public AbstractBaseArchiverOnFile(final String baseDirectory) {
+        this(baseDirectory, 0);
+    }
+
+    public AbstractBaseArchiverOnFile(final String baseDirectory, final long chunkMillis) {
+        this.baseDirectory = baseDirectory;
+        this.chunkMillis = chunkMillis;
+    }
+
+    public Path getFile(final String fileNameFormat, final Object... args) {
+        return Paths.get(this.baseDirectory, this.subDirectory, String.format(fileNameFormat, args));
+    }
+
+    public void setSubDirectory(final String subDirectory) {
```

Comment:
Does this have to be public?

Reply:
This one actually isn't used so technically no, but it's meant for implementation that might use the file archive and wants to break things up into multiple directories. I think it will be (and was during testing) a good feature. If you want to remove it we can, but otherwise it should probably stay public.